src/main/java/com/github/maxopoly/MemeMana/model/MemeManaPouch.java

```diff
@@ -28,9 +37,13 @@ public void cleanupPouch() {
 		}
 	}
 
-	public void addNewUnit(int id, double amount) {
+	public void addNewUnit(MemeManaUnit unit) {
```

Comment:
I didnt finish it up, but I intended to make it so MemeManaUnits can't be instanciated outside of the package (meaning it's constructor would have the default visibility). I'm not too tied to that idea and it might not be ideal anyway, because it's probably easier to instanciate MemeManaUnits in the DAO class than to call a method on a poach every time one is loaded from the DB. 

So basically this is fine, but just wanted to note why this was the way it was.

Reply:
The problem with the (id,amount) constructor is that it gets the system time, so we would have to have database access in the default comstuctor, which not something we want with the constructor exposed.